app/controllers/NotificationController.java

```diff
@@ -1,25 +1,32 @@
 package controllers;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import models.Notification;
+import models.enums.NotificationType;
 import org.apache.commons.lang3.exception.ExceptionUtils;
+import play.mvc.BodyParser;
 import play.mvc.Controller;
 import play.mvc.Result;
 import repository.interfaces.NotificationRepository;
 
 import javax.inject.Inject;
 import java.sql.Timestamp;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import static play.libs.Json.toJson;
 
 public class NotificationController extends Controller {
     private final NotificationRepository notificationRepository;
 
     @Inject
     public NotificationController(NotificationRepository notificationRepository) { this.notificationRepository = notificationRepository; }
 
-    public CompletionStage<Result> getNotifications() {
-        return notificationRepository.getAllNotifications().thenApplyAsync(notifications -> {
-            return ok(notifications);
-        }).exceptionally(ex -> {
-            Result res = null;
+    public CompletionStage<Result> getNotifications(long stageId) {
+        return notificationRepository.getAllNotifications(stageId).thenApplyAsync(notifications -> ok(toJson(notifications.collect(Collectors.toList())))).exceptionally(ex -> {
+            Result res;
             switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
                 case "IndexOutOfBoundsException":
                     res = badRequest("No notifications are set in DB.");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "No notifications are set in DB." 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)